### ISSUE TITLE
모델 선정을 위한 모델 테스트

### ISSUE THREAD

**Issue by xown3197** [sorachin/AI_Team10#16]
### 기준
    - 접근성 : 모델의 정보를 많이 얻을 수 있는가?
    - 정확도 : 현 데이터와 쌓은 계층으로 어떤 모델이 더 나은 정확도를 보여주는가?
- 선정 기준 추가 : 텀 프로젝트와 같은 한식 데이터 셋으로 더 성능이 나은 모델을 분석한 논문: [한식 이미지 분류에서의 미리 학습 된 컨볼루셔널 뉴럴 네트워크 간 성능 비교 분석](https://www.eiric.or.kr/community/post2_cseric.php?m=view&gubun=201804&num=15137&pg=5&seGubun=10&seGubun1=&SnxGubun=%C6%F7%BD%BA%C5%CD&searchBy=&searchWord=)을 발견하여 참고하여 선정
      1. DenseNet 82.17%
      2. ResNet 81.05%
      3. GoogLeNet 78.38%
      4. VGGNET 33.43%

### 접근성 선정 방법
    1. 딥러닝에 검색했을 때 많이 나오는 모델
    2. keras 라이브러리에 삽입되어있는 모델
    3. 추가 가산점 : tensorflow hub에 있는 모델
### 접근성을 기준으로 하여 선정된 모델 후보
- ResNet
- GoogleNet ~ Inception V3 
- Xception : 대규모 학습에 어울리는 모델이기에 탈락 + tf hub에 없음
- Mobilenet
- Densnet : tf hub에 없기에 탈락
- VGGNet : 성능이 낮아 탈락
### 1차 선정된 모델
- ResNet, Inception V3, Mobilenet

### 정확도 선정 방법
    - tensorflow hub에 학습된 모델이 있는 경우(pre training으로 정확도 향상과 학습 속도 향상)
    - 모델 테스트를 통한 분석
        - 코랩에서 학습으로 통한 정확도 비교 
        - 텐서보드를 통한 정확도 비교
#### 모델 테스트에서 사용된 명세
- 입력 데이터 : food small data (30의 클래스, 클래스 당 300장)
- pre-training (feature extraction) 사용 

|  <center> Pre_training Model </center> |  <center>ResNet</center> |  <center>Inception V3</center> | <center>Mobilenet</center> |
|:--------:|:---------:|:-------:|:---------:|
|**학습 데이터** | <center>ImageNet </center> | = | = |
|**FC** | <center>Dens(30, softmax)</center> | = | = |
|**epoch** | <center>1 </center> | = | = |
|**learning_rate**| 0.01| = | = |
|**step**| 282| = | = |
|**Gradient**| Adam| = | = |
|**Loss**|categorical entropy| = | = | 
|**Acc**| 0.8214| 0.6962 | 0.7143|

- [ResNet](https://colab.research.google.com/drive/14sopcreD6tYC0ZyTb3dgmNlLlHQJx_Lj)
- [Inception V3](https://colab.research.google.com/drive/1JUhA2Pnm0xh7M7Eu_hKzatl50jCFTc5k)
- [Mobilenet](https://colab.research.google.com/drive/1zbZbnMHSnb4JqNVoABpKeAlocjtLiJqY )

### ResNet으로 선정
- 유명한 모델이기에 접근성이 높고, 사용하는 라이브러리에 내장되어 있는 모델
- 간략한 모델 테스트 결과, 1차 선정된 모델 중 가장 성능이 좋기에 최종 선정

------------------------------------------------------------------------------------------------------------------
### pre-traing 모델을 hub를 통해 받아 왔지만, [모델 설계 변경](https://github.com/sorachin/AI_Team10/issues/23)으로 keras를 통해 trained model을 불러옴으로써, DenseNet 사용이 가능해짐
